### ISSUE TITLE
Version solving system implementation issue

### ISSUE THREAD

**Issue by vsedyshev-intermedia** [python-poetry/poetry#7268]
Poetry version: 1.1.15

Since using `poetry` with GitHub source code to resolve cascade dependencie panic with next output:

```
  SolverProblemError

  Because H (rev 4.5.0) depends on P (tag 4.8.0)
   and M depends on P (tag 4.8.0), H is forbidden.
  So, because M depends on H (tag 4.5.0), version solving failed.
```

Let's see version resolving charts:

```
  M (main)
  +--P (tag=4.8.0)
  +--H (tag=4.5.0)
     +--P (tag=4.8.0)
```

Look P is same in our cases but provided as source code and `poetry` solve as different version.

**Comment by neersighted** [comment]
This is for an old version of Poetry that we are not providing support for, and you have not provided a reproduction or followed the issue template. There's not much we can do to help you with this little information; please update to a current version of Poetry and follow the issue template so we can help you if that does not solve your issue. There have been numerous fixes to this code since 1.1.

**Comment by vsedyshev-intermedia** [comment]
Since `poetry` version 1.2 and 1.3 provide next open issue with keyring at #1917
it require additional time to switch on modern version.

Since I more deep dive in issue I found that GitHub source repository describe with different scheme "git+ssh://" and "ssh://". Could you please suggest what right way of specify source bas packages with?
